Source/Core/GeometryPipeline.js

```diff
@@ -789,11 +792,29 @@ define([
         //>>includeEnd('debug');
 
         var modelMatrix = instance.modelMatrix;
+        var boundingSphere = instance.geometry.boundingSphere;
+        var hasBS = defined(boundingSphere);
+
+        if (hasBS && defined(instance.attributes.offset)) {
+            var offset = Cartesian3.fromArray(instance.attributes.offset.value, 0, instanceOffsetScratch);
+            var extend = defined(instance.geometry.attributes) && defined(instance.geometry.attributes.applyOffset) && instance.geometry.attributes.applyOffset.values.indexOf(0) !== -1;
+            if (extend) {
+                var origBS = BoundingSphere.clone(boundingSphere, offsetBoundingSphereScratch1);
+                var offsetBS = BoundingSphere.clone(boundingSphere, offsetBoundingSphereScratch2);
+                offsetBS.center = Cartesian3.add(offsetBS.center, offset, offsetBS.center);
+                boundingSphere = BoundingSphere.union(origBS, offsetBS, boundingSphere);
```

Comment:
Keep the original bounding sphere and have the primitive update them after creation. The union here could cause the bounding sphere to be huge and we could lose benefits from culling.